### PR TITLE
BUG: pd.array raising with NumPy array and large dtype

### LINKED ISSUE #52590: BUG: pd.array raising for large_string/large_binary with NumPy array
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
arr = np.array(["a", "b"])
pd.array(arr, dtype=pd.ArrowDtype(pa.large_string()))
```

```
arr = np.array([b"a", b"b"])
pd.array(arr, dtype=pd.ArrowDtype(pa.large_binary()))
```

```
Traceback (most recent call last):
  File "/Users/patrick/Library/Application Support/JetBrains/PyCharm2023.1/scratches/scratch.py", line 530, in <module>
    pd.array(arr, dtype=pd.ArrowDtype(pa.large_string()))
  File "/Users/patrick/PycharmProjects/pandas/pandas/core/construction.py", line 327, in array
    return cls._from_sequence(data, dtype=dtype, copy=copy)
  File "/Users/patrick/PycharmProjects/pandas/pandas/core/arrays/arrow/array.py", line 250, in _from_sequence
    scalars = pa.array(scalars, type=pa_dtype, from_pandas=True)
  File "pyarrow/array.pxi", line 316, in pyarrow.lib.array
  File "pyarrow/array.pxi", line 83, in pyarrow.lib._ndarray_to_array
  File "pyarrow/error.pxi", line 121, in pyarrow.lib.check_status
pyarrow.lib.ArrowNotImplementedError: NumPyConverter doesn't implement <large_string> conversion. 
```

### Issue Description

This raises since NumPy does not implement the converter. I think we should work around this.

### Expected Behavior

Returns an ArrowExtensionArray

### Installed Versions

<details>

main
</details>

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #52590 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

not sure if we want to backport this though...

### CODE REVIEW

File: pandas/core/arrays/arrow/array.py [hunk 1]
```diff
@@ -241,6 +241,15 @@ def _from_sequence(cls, scalars, *, dtype: Dtype | None = None, copy: bool = Fal
...
         """
         pa_dtype = to_pyarrow_type(dtype)
+        if (
+            isinstance(scalars, np.ndarray)
+            and isinstance(dtype, ArrowDtype)
+            and (
+                pa.types.is_large_binary(pa_dtype) or pa.types.is_large_string(pa_dtype)
+            )
+        ):
+            scalars = scalars.tolist()
```
Comment (mroeschke) [inline pandas/core/arrays/arrow/array.py:251]: Could you open an issue in arrow and link that issue here?

Comment (phofl) [reply pandas/core/arrays/arrow/array.py:256]: Done

**Comment by phofl** [comment]
This should be ready

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @phofl

**Pull request merged**